dd-sdk-android/src/main/kotlin/com/datadog/android/log/Logger.kt

```diff
@@ -466,5 +468,21 @@ internal constructor(private val handler: LogHandler) {
         tags.remove(tag)
     }
 
+    private fun safelyAddAttribute(key: String, value: Any?) {
+        value?.let {
+            attributes[key] = value
+        }
+    }
+
+    private fun safelyRemoveTagsWithKey(keyFilter: (String) -> Boolean) {
+        // we first gather all the objects we want to remove based on a copy
+        val toRemove = tags
+            .toArray()
+            .filter {
+                it is String && keyFilter(it)
```

Comment:
in theory it is always string ! 
Why can't you just use `tags.filter(keyFilter)` ?

Reply:
if you do that will not be thread safe anymore. You need to use `toArray` function to make a copy because otherwise another thread might modify the `collection` while you are filtering (ConcurrentModificationException). Problem is that `toArray` returns an Object[] for some reason and I need it to cast it to a String there. The method I am using after `removeAll` it is already Thread safe on at the collection level (it uses a synchronized block).

Reply:
Okay, makes sense, but then you can use Kotlin's `toTypedArray()` instead, it will turn it into a `String[]`